config/initializers/after_sign_in.rb

```diff
@@ -0,0 +1,8 @@
+Warden::Manager.after_set_user except: :fetch do |user, _auth, _opts|
+  if user&.member? && ! user.membership_current?
```

Comment:
This should be a method in the User model so that (a) the logic and functionality for updating the state of the User (the membership state, in this case) is in the model.  In the future, when we need to make this richer (perhaps do more checks or do more changes when the conditions are met), we will not be prone to forgetting that this code is tucked away in the initializers.
(Also:  this really _is_ the responsibility of the User model as we have it right now.  Although I can certainly see a future where we have a class whose responsibility it is to review and update the membership status)